src/main/java/heronarts/lx/midi/surface/MidiFighter64.java

```diff
@@ -0,0 +1,203 @@
+  public static final int LED_HOT_PINK_DIM = 59;
+  public static final int LED_GOLDENROD = 62;
+  public static final int LED_LAWN_GREEN = 63;
+  // There are 64 more colors, but they seem like dupes.
+  // There might be a small number of slightly unique ones in that range.
+
+  // Holds the colors that will be sent out to the device. This array
+  // starts with (0,0), (0,1), ... (7,7) of the LEFT page, and then
+  // the same ordering for the RIGHT page.
+  private int[] colors;
+
+  // Converts a MIDI note from the MF64 into information about which
+  // button was pressed
+  public class Mapping {
+    // If an unexpected note is received, this is set to false
+    // and all other attributes should be ignored.
+    public boolean valid;
+    // The MF64 has left and right virtual button pages.
+    public enum Page {
+      LEFT, RIGHT;
+    }
+    public Page page;
+    // 0 is the bottom row, 7 the top.
+    public int row;
+    // 0 is the left column, 7 the right
+    public int col;
+
+    public Mapping(MidiNoteOn note) {
+      int pitch = note.getPitch();
+      int channel = note.getChannel();
+      this.valid = true;
+      if (channel == 2) {
+        this.page = Page.LEFT;
+      } else if (channel == 1) {
+        this.page = Page.RIGHT;
+      } else {
+        LX.warning("Got wild-channel MIDI note " + note);
+        this.valid = false;
+        return;
+      }
+
+      if (pitch >= 36 && pitch <= 67) {
+        this.row = (pitch / 4) - 9;
+        this.col = pitch % 4;
+      } else if (pitch >= 68 && pitch <= 99) {
+        this.row = (pitch / 4) - 17;
+        this.col = pitch % 4 + 4;
+      } else {
+        LX.warning("Got wild-pitch MIDI note " + note);
+        this.valid = false;
+      }
+    }
+  }
+
+  public void setColors(int[] newColors) {
+    if (newColors.length != 128) {
+      throw new IllegalArgumentException("newColors is of length " + newColors.length);
+    }
+    for (int i = 0; i < 128; i++) {
+      if (newColors[i] < 0 || newColors[i] > 127) {
+        throw new IllegalArgumentException("newColors contained value " + newColors[i]);
+      }
+    }
+    System.arraycopy(newColors, 0, this.colors, 0, 128);
+    this.sendColors();
+  }
+
+  private void sendColors() {
+    int ci = 0;
+    for (int channel = 2; channel >= 1; channel--) {
+      for (int i = 0; i < 64; i++) {
+        this.sendNoteOn(channel, pitchFromXY[i], this.colors[ci++]);
+      }
+    }
+  }
+
+  public MidiFighter64(LX lx, LXMidiInput input, LXMidiOutput output) {
+    super(lx, input, output);
+    this.colors = new int[8*8*2];
+    Arrays.fill(this.colors, 0, 64, LED_WARM_WHITE);
+    Arrays.fill(this.colors, 64, 128, LED_RED_DIM);
+  }
+
+  @Override
+  protected void onEnable(boolean on) {
+    this.sendColors();
+  }
+
+  @Override
+  protected void onReconnect() {
+    if (this.enabled.isOn()) {
+      initialize(true);
+    }
+  }
+
+  private void initialize(boolean reconnect) {
```

Comment:
You don't call initialize in the base case, so I might actually ditch this method and just put the `sendColors()` call directly in the reconnect case. Otherwise this method's naming/existence is potentially confusing, as there's no actual first-time initialization.